src/com/lftechnology/java/training/alina/jdbc/dao/user/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+        List<Employee> employeeDetails = employeeDao.findByPk(userId);
+        for (Employee employee : employeeDetails) {
+            userService.getEmployeeRole(scanner, employee);
+        }
+    }
+
+    @Override
+    public List<User> findByPk(Integer userId) {
+        List<User> userList = new ArrayList<User>();
+        Map<Integer, Object> params = new HashMap<Integer, Object>();
+        params.put(1, userId);
+        Database database = UserService.setDatabaseParams(params);
+        String sql = "Select * from user where user_id=?) ";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                UserService userService = new UserService();
+                userList.add(userService.map(result));
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
+        }
+        return userList;
+    }
+
+    @Override
+    public List<User> findAll() {
+        List<User> userList = new ArrayList<User>();
+        User user = new User();
+        String sql = "Select * from user";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                user.setUsername(result.getString("username"));
+                user.setTerminated(result.getBoolean("is_terminated"));
+                user.setCreatedAt(result.getTimestamp("created_at"));
+                userList.add(user);
+            }
+            preparedStatement.close();
+        } catch (SQLException e) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { e });
+        }
+        return userList;
+    }
+
+    @Override
+    public User addNew(User user) {
+        String sql = "Insert into user (username,password,is_terminated,created_at) values (?,?,?,?)";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, user.getUsername());
+            preparedStatement.setString(2, user.getPassword());
+            preparedStatement.setBoolean(3, Constants.NOT_TERMINATED);
+            preparedStatement.setTimestamp(4, user.getCreatedAt());
+            int result = preparedStatement.executeUpdate();
+            ResultSet rs = preparedStatement.getGeneratedKeys();
+            if (result > 0) {
+                if (rs.next()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)